at_client/src/main/java/org/atsign/client/util/KeysUtil.java

```diff
@@ -18,68 +18,93 @@
+            file = new File(rootFolder);
+            if (!file.exists()) {
+                // If atKeys file does not exist,
+                // create dir in ~/ and request
+                // to store key files there
+                _makeRootFolder();
+            }
+            System.out.println("Saving keys to " + file.getAbsolutePath());
+        } else {
+            System.out.println("Saving keys to " + file.getAbsolutePath());
+            // noinspection ResultOfMethodCallIgnored
             file.delete();
         }
 
         String selfEncryptionKey = keys.get(selfEncryptionKeyName);
 
         Map<String, String> encryptedKeys = new TreeMap<>();
 
-        // We encrypt all the keys with the AES self encryption key (which is left unencrypted)
+        // We encrypt all the keys with the AES self encryption key (which is left
+        // unencrypted)
         encryptedKeys.put(selfEncryptionKeyName, selfEncryptionKey);
-        encryptedKeys.put(pkamPublicKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(pkamPublicKeyName), selfEncryptionKey));
-        encryptedKeys.put(pkamPrivateKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(pkamPrivateKeyName), selfEncryptionKey));
-        encryptedKeys.put(encryptionPublicKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPublicKeyName), selfEncryptionKey));
-        encryptedKeys.put(encryptionPrivateKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPrivateKeyName), selfEncryptionKey));
+        encryptedKeys.put(pkamPublicKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(pkamPublicKeyName), selfEncryptionKey));
+        encryptedKeys.put(pkamPrivateKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(pkamPrivateKeyName), selfEncryptionKey));
+        encryptedKeys.put(encryptionPublicKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPublicKeyName), selfEncryptionKey));
+        encryptedKeys.put(encryptionPrivateKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPrivateKeyName), selfEncryptionKey));
 
         String json = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(encryptedKeys);
         Files.write(file.toPath(), json.getBytes(StandardCharsets.UTF_8));
     }
 
-    private static File getKeysFile(AtSign atSign) {
-        return new File(System.getProperty("user.dir") + "/keys/" + atSign + "_key.atKeys");
-    }
-
     public static Map<String, String> loadKeys(AtSign atSign) throws Exception {
-        _makeRootFolder();
-        File file = getKeysFile(atSign);
-        if (! file.exists()) {
-            throw new AtException("loadKeys: No file at " + file.getAbsolutePath());
+        File file = new File(rootFolder);
+        if (!file.exists()) {
+            // if keys do not exist in project dir, search in ~/
+            rootFolder = System.getProperty("user.home") + "/.atsign/keys/" + atSign + "_key.atKeys";
```

Comment:
Same problem as with use of rootFolder in the saveKeys method. Use a local variable called `keysFileName` instead of modifying rootFolder variable which should be constant